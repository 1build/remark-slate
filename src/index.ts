import plugin, { transform, defaultNodeTypes } from './serializer';

export { transform, defaultNodeTypes };

export default plugin;
