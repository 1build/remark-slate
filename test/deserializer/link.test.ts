import { transform } from '../../src';

it('transforms link', () => {
  expect(
    transform({
      type: 'link',
      url: 'https://jackhanford.com',
      children: [{ value: 'Personal website' }],
    })
  ).toMatchSnapshot();
});
