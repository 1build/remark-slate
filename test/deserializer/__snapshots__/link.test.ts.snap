// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms link 1`] = `
Object {
  "children": Array [
    Object {
      "text": "Personal website",
    },
  ],
  "link": "https://jackhanford.com",
  "type": "link",
}
`;
